janus_core/helpers/correlator.py

```diff
@@ -178,6 +178,8 @@ class Correlation:
 
     Parameters
     ----------
+    n_atoms : int
+        Number of possible atoms to track.
```

Comment:
It's not entirely clear to me from this how this works?

Would you not expect all atoms to be tracked by default, and otherwise would you not want to specify something through something like a mask, rather than just a number (the first n atoms?).

I may be misunderstanding its purpose though.

Reply:
This annoys me but I don't see a way around. It is not the mask, but merely to apply a mask that may have an unspecified number of atoms that are resolved to the number of atoms in md. 

The correlation needs to know how many correlators to spawn (i.e. from the mask combined with an actual atoms object) for averaging over atoms. And these need to be in a set order for correct averaging over both dimensions, x, y, x and atoms.

I have added "masks" to the observables, i.e. the Velocity observable, in the same ways as post_process using ```SliceLike``` which allows for ```unspecified endings``` (```start, stop, step``` = ```5, None, 1```). This leaves them potentially not tied to a specific max atom count (ignoring bounds issues). They could have one eliminating this, but then the user will have to specify say an atoms object or count when creating a ```Velocity``` observable to set that. Perhaps that is ok.

Currently this value, ```n_atoms```, is passed when the Correlation is created, or in other words when it is attached to a MD object that has some atoms, and in particular a definite ```n_atoms```. So that the mask can be resolved into a specific number of correlators to be spawned. We could just spawn ```n_atoms``` always but that may be inefficient in some cases.